src/NetTopologySuite.Lab/Hull/ChiShape.cs

```diff
@@ -0,0 +1,151 @@
+                        Debug.WriteLine(neighbour);
+                        CheckMaxEdgeAndListBorderTriangles(quadEdgeTriangleList, neighbour);
+                    }
+                }
+            }
+            var lineMerger = new LineMerger();
+            var fact = new GeometryFactory();
+            foreach (var triangle in tris)
+            {
+                if (!triangle.IsLive()) continue;
+                for (int i = 0; i < 3; i++)
+                {
+                    if (triangle.GetAdjacentTriangleAcrossEdge(i) == null || !triangle.GetAdjacentTriangleAcrossEdge(i).IsLive())
+                    {
+                        lineMerger.Add(triangle.GetEdge(i).ToLineSegment().ToGeometry(fact));
+                    }
+                }
+            }
+            var mergedLine = (LineString)lineMerger.GetMergedLineStrings().FirstOrDefault();
+            if (mergedLine.IsRing)
+            {
+                var lr = new LinearRing(mergedLine.CoordinateSequence, fact);
+
+                var concaveHull = new Polygon(lr, null, fact);
+
+                return concaveHull;
+            }
+            return mergedLine;
+        }
+
+
+
+        private void CheckMaxEdgeAndListBorderTriangles(SortedList<QuadEdge, QuadEdgeTriangle> quadEdgeTriangleList, QuadEdgeTriangle triangle)
+        {
+            bool canBeRemoved = RemovalGivesRegularPolygon(triangle);
+            if (canBeRemoved)
+            {
+                for (int i = 0; i < 3; i++)
+                {
+                    if (triangle.GetAdjacentTriangleAcrossEdge(i) == null)
+                    {
+                        var edge = triangle.GetEdge(i);
+                        if (!quadEdgeTriangleList.ContainsKey(edge))
+                        {
+                            Debug.WriteLine($"edge: {edge} index: {i} triangle: {triangle}");
+                            quadEdgeTriangleList.Add(edge, triangle);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static bool RemovalGivesRegularPolygon(QuadEdgeTriangle triangle)
+        {
+            int neighbourCount = 0;
+            var neighbours = triangle.GetNeighbours();
+            for (int i = 0; i < 3; i++)
+            {
+                if (neighbours[i] != null && neighbours[i].IsLive())
+                {
+                    neighbourCount++;
+                }
+            }
+            bool canBeRemoved = neighbourCount == 2;
+            return canBeRemoved;
+        }
+
+        private QuadEdgeSubdivision BuildDelaunay()
+        {
+            var builder = new DelaunayTriangulationBuilder();
+            builder.SetSites(_geom);
+            var subDiv = builder.GetSubdivision();
+            return subDiv;
+        }
+    }
+
+    internal class QuadEdgeComparer : Comparer<QuadEdge>
+    {
+        public override int Compare(QuadEdge x, QuadEdge y)
+        {
+            return x.Length.CompareTo(y.Length);
```

Comment:
The `IComparer<TKey>` used for a `SortedList<TKey, TValue>` is supposed to give a total ordering.  This only checks the `Length`, which means that if two different `QuadEdge`s happen to have the same length, then only one of them will be allowed to live in the list at a given time.

Reply:
Do we need to solve this edge case :)

Reply:
Can you help me with creating a test for this, not sure when this would be an issue.